ui/src/components/NewWorkflow/HTTPTask.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2021 Chaos Mesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { Form, Formik } from 'formik'
+import { FormControlLabel, Switch } from '@material-ui/core'
+import { MenuItem, Typography } from '@material-ui/core'
+import { SelectField, Submit, TextField } from 'components/FormField'
+import { Template, TemplateType } from 'slices/workflows'
+import { parseHTTPTask, renderHTTPTask } from 'api/workflows'
+import { useEffect, useRef, useState } from 'react'
+
+import Paper from 'components-mui/Paper'
+import PaperTop from 'components-mui/PaperTop'
+import { RequestForm } from 'api/workflows.type'
+import Space from 'components-mui/Space'
+import T from 'components/T'
+import { makeStyles } from '@material-ui/styles'
+import { useIntl } from 'react-intl'
+import { validateName } from 'lib/formikhelpers'
+
+const useStyles = makeStyles({
+  field: {
+    width: 180,
+  },
+})
+
+const HTTPMethods = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'HEAD', 'OPTION']
+
+interface HTTPTaskProps extends CommonTemplateProps {
+  childrenCount: number
+  submitTemplate: (template: Template) => void
+}
+
+interface CommonTemplateProps {
+  name?: string
+  deadline?: string
+  type: TemplateType
+  templates: Template[]
+  externalTemplate?: Template
+}
+
+const HTTPTask: React.FC<HTTPTaskProps> = (props) => {
+  const intl = useIntl()
+  const classes = useStyles()
+
+  const { submitTemplate } = props
+  const onSubmit = (form: RequestForm) => {
+    renderHTTPTask(form)
+      .then((response) => {
+        const { name, task } = response.data!
+        const { container } = task
+        const result: Template = {
+          name,
+          children: [],
+          type: TemplateType.Custom,
+          custom: {
+            container,
+            conditionalBranches: [],
+          },
+        }
+        submitTemplate(result)
+      })
+      .catch(console.error)
+  }
+  const formRef = useRef<any>()
+  const [initialValues, setInitialValues] = useState<RequestForm>({
+    name: props.name || '',
+    url: '',
+    method: '',
+    body: '',
+    followLocation: false,
+    jsonContent: false,
+  })
+
+  const validateRequestForm = (newValue: RequestForm) => {
```

Comment:
It looks like this validation is useless, I guess it can be removed?

Reply:
done in 8efaaa00e506f23c92288e0b4e128eabe3f700a5